cmd/ocagent/exparser/datadog.go

```diff
@@ -51,21 +51,29 @@ func (d *datadogExporter) MakeExporters(config []byte) (se view.Exporter, te tra
 	if err := yaml.Unmarshal(config, &c); err != nil {
 		log.Fatalf("Cannot unmarshal data: %v", err)
 	}
-	if c := c.Datadog; c.EnableMetrics || c.EnableTraces {
-		// TODO(jbd): Create new exporter for each service name.
-		de := datadog.NewExporter(datadog.Options{
-			Namespace: c.Namespace,
-			TraceAddr: c.TraceAddr,
-			StatsAddr: c.MetricsAddr,
-			Tags:      c.Tags,
-		})
-		if c.EnableMetrics {
-			se = de
-		}
-		if c.EnableTraces {
-			te = de
-		}
-		closer = de.Stop
+
+	dc := c.Datadog
+	if dc == nil {
+		return nil, nil, nil
+	}
+	enableAnyExporter := dc.EnableMetrics || dc.EnableTraces
+	if !enableAnyExporter {
+		return nil, nil, nil
+	}
+
+	// TODO(jbd): Create new exporter for each service name.
+	de := datadog.NewExporter(datadog.Options{
+		Namespace: dc.Namespace,
+		TraceAddr: dc.TraceAddr,
+		StatsAddr: dc.MetricsAddr,
+		Tags:      dc.Tags,
+	})
+	if dc.EnableMetrics {
```

Comment:
Do we need this now?

Reply:
@songy23 that was just translating the old code :)